### PR TITLE
Create resource timing entries for fetch network errors

### PR DESCRIPTION
**Pull request by noamr**
An initial tests for https://github.com/whatwg/fetch/issues/1215

Added test cases for several network error scenarios. The test cases verify that a resource timing entry is created for those resources, without the "internal" timing info (redirects, networking), as if the resource was cross-origin no-cors and lacked TAO.

The error test cases include:
- CORS fail (which is also covered in existing TAO tests)
- about URLs
- invalid blobs
- Posting to blobs
- Only-if-cached header for a fresh resource
- Abort HTTP/blob request
- Invalid scheme
- Too many redirects
- Invalid certificate
- Host not found
- CSP blocked

These tests come in conjunction with https://github.com/whatwg/fetch/pull/1311

### CODE REVIEW

File: resource-timing/resources/entry-invariants.js [hunk 1]
```diff
@@ -481,3 +481,27 @@ const attribute_test = (loader, path, run_test, test_label) => {
...
+          () => fetch(url, args_or_loader);
+        try {
+          url = await loader()
+        } catch (e) {
+            const timeAfter = performance.now();
+            const entries = performance.getEntriesByName(url);
+            assert_equals(entries.length, 1, 'resource timing entry for network error');
+            const entry = entries[0]
+            assert_equals(entry.startTime, entry.fetchStart, 'startTime and fetchStart should be equal');
+            assert_greater_than(entry.startTime, timeBefore, 'startTime and fetchStart should be greater than the time before fetching');
```
Comment (yoavweiss) [inline resource-timing/resources/entry-invariants.js:499]: Should that be "greater_than_equal"?

**Review by yoavweiss — COMMENTED** [review]
Thanks!! A few comments

File: resource-timing/csp-block-fetch.js.headers [hunk 2]
```diff
@@ -0,0 +1 @@
+Content-Security-Policy: connect-src 'none';
```
[the remark this answers is not in the document]
Comment (noamr) [reply resource-timing/csp-block-fetch.js.headers:1]: Right, debris from copy/paste

**Review by yoavweiss — APPROVED** [review]
LGTM

**Comment by noamr** [comment]
> LGTM

Thanks!
Should probably wait until the spec change is approved and merge at the same time.

**Comment by noamr** [comment]
Modified test to include only 5 scenarios for now:
- CORS failure
- DNS failure
- Timeout (IP address not found)
- Only-if-cached-resource that wasn't cached
- Too many redirects

Currently Chrome & Firefox are inconsistent around which of these create a resource timing entry.
Safari does not create RT entries for network errors.

**Comment by noamr** [comment]
This actually catches a Firefox bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1523275

File: resource-timing/entries-for-network-errors.sub.https.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  controller.abort();
+  const res = fetch(url, { signal });
+  return res
+}
+
+network_error_entry_test(
+    `${REMOTE_ORIGIN}${validXmlUrl}`, null, "failed cross-origin requests");
+
+network_error_entry_test(`/common/redirect.py?location=${validDataURL}`, null, "non-HTTP redirect");
+network_error_entry_test('//{{hosts[][nonexistent]}}/common/dummy.xml', null, "DNS failure");
```
Comment (yoavweiss) [inline resource-timing/entries-for-network-errors.sub.https.html:31]: Mixed content could be another interesting case

File: resource-timing/entries-for-network-errors.sub.https.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+<script>
+const {REMOTE_ORIGIN} = get_host_info();
+const validXmlUrl = '/common/dummy.xml';
+
+function fetchAndAbort(url) {
+  const controller = new AbortController();
+  const signal = controller.signal;
+  controller.abort();
+  const res = fetch(url, { signal });
+  return res
```
Comment (yoavweiss) [inline resource-timing/entries-for-network-errors.sub.https.html:24]: Nit: ";"

**Review by yoavweiss — APPROVED** [review]
LGTM % nit and suggestion for another test

**Comment by noamr** [comment]
The Firefox failures are legit, e.g. https://bugzilla.mozilla.org/show_bug.cgi?id=1523275. How does this get merged when it fails firefox nightly?

**Comment by noamr** [comment]
> LGTM % nit and suggestion for another test

Added a test for mixed content.

**Pull request merged**